source/agora/common/BlockStorage.d

```diff
@@ -359,12 +360,33 @@ public class BlockStorage
 
     ***************************************************************************/
 
-    private ubyte[] read (size_t from, size_t to) @trusted
+    private ubyte[] read (size_t from, size_t to) nothrow @trusted
     {
-        assert(this.file !is null);
-        assert(this.getFileLength >= to);
-
-        return cast(ubyte[])this.file[from .. to];
+        if (
+            (this.file !is null) &&
+            (from / MapSize == this.file_index) &&
+            (to / MapSize == this.file_index))
+        {
+            ubyte[] data;
+            try
+            {
+                const size_t x0 = from - this.file_base;
+                const size_t x1 = to - this.file_base;
+                data = cast(ubyte[])this.file[x0 .. x1];
+            }
+            catch (Exception ex)
```

Comment:
Flat-out ignoring an exception does not seem like the best way to go

Reply:
fixed